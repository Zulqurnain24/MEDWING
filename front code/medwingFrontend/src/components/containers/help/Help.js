import React, { Component } from "react";
import { View, StyleSheet, Text} from "react-native";

class Help extends Component {

    render() {

        return (
            <View nativeID={"2"} style={styles.container}>
                 <Text>
                    <Text style={ styles.boldText }>{"Instructions: \n"}</Text>
                    <Text style={ styles.bulletText }>{"1.  The app loads the pins from mlab database.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"2.  You can add further pins onto the map by long press on the map.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"3.  You can touch the markers on the map and drag to alter their location.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"4.  You can tap on the markers to select them.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"5.  You can delete the marker by tapping the delete button.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"6.  You can edit the marker by pressing the edit button and then entering the description to alter the description of the marker.  \n \n"}</Text>
                    <Text style={ styles.bulletText }>{"7.  In backend app the region setting is for Germany so only the pins within Germany will be displayed.  \n \n"}</Text>
                </Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
      marginTop:-200,
      alignItems: 'center',
      width: 400,
    },
    bulletText: {
        flex: 1,
        width: 10,
    },
    boldText: {
        fontWeight: 'bold'
    }
});

export default Help;