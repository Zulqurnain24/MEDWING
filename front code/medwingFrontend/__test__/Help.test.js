//Help Component snapshot testing
import 'react-native';
import React from 'react';
import Help from '../src/components/containers/help/Help';
import renderer from 'react-test-renderer';

test('Help snapshot', ()=>{
    let snap = renderer.create(
        <Help />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});