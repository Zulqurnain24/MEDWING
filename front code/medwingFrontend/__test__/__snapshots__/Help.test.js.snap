// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": -200,
    }
  }
>
  <Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Instructions: 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      1.  The app loads the pins from mlab database.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      2.  You can add further pins onto the map by long press on the map.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      3.  You can touch the markers on the map and drag to alter their location.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      4.  You can tap on the markers to select them.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      5.  You can delete the marker by tapping the delete button.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      6.  You can edit the marker by pressing the edit button and then entering the description to alter the description of the marker.  
 

    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "width": 10,
        }
      }
    >
      7.  In backend app the region setting is for Germany so only the pins within Germany will be displayed.  
 

    </Text>
  </Text>
</View>
`;
